src/Services/SubmitService/SubmitService.cs

```diff
@@ -79,21 +88,59 @@ public async Task<string> ProcessSubmission(MappingEntity mappingEntity, string
                 reference = answers.CaseReference;
             }
 
-            return _submissionServiceConfiguration.FakeSubmission 
-                ? ProcessFakeSubmission(mappingEntity, form, sessionGuid, reference) 
+            if (baseForm.Pages is not null && mappingEntity.FormAnswers.Pages is not null)
+            {
+                var journeyPages = baseForm.GetReducedPages(mappingEntity.FormAnswers);
+
+                var isAutoConfirm = journeyPages.Any(_ => _.Elements.Any(_ => _.Type.Equals(EElementType.Booking) && _.Properties.AutoConfirm.Equals(true)));
+                if (isAutoConfirm)
+                    return await ConfirmBooking(mappingEntity, form, sessionGuid, baseForm, reference);
+            }
+
+            return _submissionServiceConfiguration.FakeSubmission
+                ? ProcessFakeSubmission(mappingEntity, form, sessionGuid, reference)
                 : await ProcessGenuineSubmission(mappingEntity, form, sessionGuid, baseForm, reference);
         }
 
+        private async Task<string> ConfirmBooking(MappingEntity mappingEntity, string form, string sessionGuid, FormSchema baseForm, string reference)
+        {
+            var bookingProperties = baseForm.Pages
+                               .Where(page => page.Elements is not null)
+                               .SelectMany(page => page.Elements)
+                               .First(element => element.Type.Equals(EElementType.Booking)).Properties;
+
+            var bookingId = Convert.ToString(mappingEntity.FormAnswers.AllAnswers
+                .ToDictionary(x => x.QuestionId, x => x.Response)
+                .Where(_ => _.Key.Contains(BookingConstants.RESERVED_BOOKING_ID))
+                .FirstOrDefault().Value);
+
+            var appointmentType = bookingProperties.AppointmentTypes
+                .GetAppointmentTypeForEnvironment(_environment.EnvironmentName);
+
+            await _bookingProviders
+                .Get(bookingProperties.BookingProvider)
+                .Confirm(new()
+                {
+                    BookingId = new Guid(bookingId),
+                    AdditionalInformation = string.Empty,
+                    OptionalResources = appointmentType.OptionalResources
+                });
+
+            return bookingProperties.BookingProvider.Equals(BookingConstants.FAKE_PROVIDER)
+                ? ProcessFakeSubmission(mappingEntity, form, sessionGuid, reference)
+             : bookingId;
```

Comment:
Can we do the check to see if this is a Fake Submission earlier in the method e.g. after line 107? This would stop us doing any complex LINQ queries that may not be needed.